src/main/java/co/unruly/matchers/function/DescriptionUtils.java

```diff
@@ -0,0 +1,14 @@
+package co.unruly.matchers.function;
+
+import static java.util.Arrays.binarySearch;
+
+public final class DescriptionUtils {
```

Comment:
This is a candidate to be moved to an "internal utils" package. I was not able to encapsulate it to only `co.unruly.matchers.function`, as it is also used by the `Java8Matchers` class. (It is also not tied in any way to "functions" per se, so it is a bit artificial to keep it in the `function` package. However, I didn't want to place it in `co.unruly.matchers`, as it is really not part of the public API of the library, and it would introduce a circular package dependency between `co.unruly.matchers` and `co.unruly.matchers.function`.

Maybe move it to `co.unruly.matchers.internal` to make it more clear that this is part of the library's internal code? Other libraries use this strategy to "tag" classes as internal, e.g. [Mockito](https://github.com/mockito/mockito/tree/release/2.x/src/main/java/org/mockito/internal)